org.eclipse.dawnsci.slicing.api/src/org/eclipse/dawnsci/slicing/api/ServiceHolder.java

```diff
@@ -0,0 +1,29 @@
+/*-
+ * Copyright 2017 Diamond Light Source Ltd.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.dawnsci.slicing.api;
+
+import org.eclipse.dawnsci.nexus.INexusFileFactory;
+
+public class ServiceHolder {
+
+	private static INexusFileFactory nexusFileFactory;
+
+	public ServiceHolder() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Hide this public constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)